File: recyclerlayout/src/main/java/com/angel/adapter/SWDiffCallBack.java
package com.angel.adapter;

import android.support.v7.util.DiffUtil;

import java.util.List;

/**
 * Created by Angel on 2017/4/10.
 */
public class SWDiffCallBack<T> extends DiffUtil.Callback {

    private List<T> olddatas;
    private List<T> newDatas;

    public SWDiffCallBack(List<T> olddatas, List<T> newDatas) {
        this.olddatas = olddatas;
        this.newDatas = newDatas;
    }

    public int getOldListSize() {
        return olddatas.size();
    }

    public int getNewListSize() {
        return newDatas.size();
    }

    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return olddatas.get(oldItemPosition).equals(newDatas.get(newItemPosition));
    }

    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return olddatas.get(oldItemPosition).equals(newDatas.get(newItemPosition));
    }
}

File: kotlinlibrary/src/main/java/com/kotlin/adapter/SWDiffCallBack.kt
package com.kotlin.adapter

import android.support.v7.util.DiffUtil

/**
 * Created by Angel on 2017/5/19.
 */
class SWDiffCallBack<T>(private val olddatas: List<T>, private val newDatas: List<T>) : DiffUtil.Callback() {

    override fun getOldListSize(): Int {
        return olddatas.size
    }

    override fun getNewListSize(): Int {
        return newDatas.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return olddatas[oldItemPosition] == newDatas[newItemPosition]
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return olddatas[oldItemPosition] == newDatas[newItemPosition]
    }
}
